tmk_core/common/action_layer.c

```diff
@@ -32,6 +33,12 @@ __attribute__((weak)) layer_state_t default_layer_state_set_kb(layer_state_t sta
  */
 static void default_layer_state_set(layer_state_t state) {
     state = default_layer_state_set_kb(state);
+    static bool has_run = false;
+    if (has_run) { 
+        eeconfig_update_default_layer(state);
+    } else {
+        has_run = true;
+    }
```

Comment:
```suggestion
    if (!has_run) { 
        eeconfig_update_default_layer(state);
        has_run = true;
    }
```
?

Also, this will only write the default layer once, until the board is restarted. Is that intentional?

Reply:
No, the goal is to only write after the first run.  The function is called on init by magic/bootmagic, so don't want to write back what was *just* read from.  

I suppose I could just do this instead. 
```suggestion
    if (has_run) { 
        eeconfig_update_default_layer(state);
    }
    has_run = true;
```

Or 
```suggestion
    has_run ? eeconfig_update_default_layer(state) : has_run = true;
```